File: Two Sum/TwoSum .py
# 48ms 58.55%
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # 使用字典来快速查找
        dic = {}
        for i in range(len(nums)):
            if nums[i] in dic:
                return [dic[nums[i]], i]
            else:
                dic[target - nums[i]] = i

File: Two Sum/Two_Sum.cpp
// 利用哈希表，线性的时间复杂度、线性的空间复杂度

class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        unordered_map<int, int> hashmap;
        vector<int> res;
        
        for (int index = 0; index < nums.size(); ++index)
        {
            if (hashmap.find(target - nums[index]) == hashmap.end())
            {
                hashmap.insert(make_pair(nums[index], index));
            }
            else
            {
                res.push_back(hashmap.at(target - nums[index]));
                res.push_back(index);
                break;
            }
        }
        return res;
    }
};